aurae-ebpf-shared/src/lib.rs

```diff
@@ -0,0 +1,8 @@
+#![no_std]
+
+#[repr(C)]
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct Signal {
+    pub signr: i32,
```

Comment:
should we use a predefined enum here? https://docs.rs/signal/0.5.1/signal/enum.Signal.html exists and i assume it supports no_std...

Reply:
Unfortunately we cannot use that from the kernel. The `Signal` crate depends on `nix` which pulls in a whole slew of transitive dependencies which cause the `aurae-ebpf` build to fail. We could have `load_tracepoint<T>` or the `PerfEventListener<T>` take an extra type parameter, which could be a more idiomatic `Rust` type that would implement `From` for the C-struct that has to traverse the FFI boundary but then again in this codepath we'd be immediately mapping it again to a `proto` type.